### ISSUE TITLE
cannot set package name with -import

### ISSUE THREAD

**Issue by godolatry** [golang/mock#49]
I'm having a problem with the -import mechanism in mockgen.  I'm trying to set the name of a package which the mock is importing.  I can't figure out how to do it.

When I create a mock with mockgen, it contains this import:

person "github.com/goblimey/films/models/person"

and the mock contains a method with a syntax error:

func (_m *MockDAO) Create(person person.Person) (person.Person, error) {
    ret := _m.ctrl.Call(_m, "Create", person)
    ret0, _ := ret[0].(person.Person)                   // syntax error here
    ret1, _ := ret[1].(error)
    return ret0, ret1
}

The syntax error is 

./MockDAO.go:76: person.Person undefined (type person.Person has no field or method Person)

This is caused by confusion between the variable person and the package person.  I need the import line to be something like:

import (
    personModel "github.com/goblimey/films/models/person"

So the import is called personModel, not person.

It looks as if I can fix this problem using -import when I run mockgen.  However, when I do that, I still get the import line that provokes the error.  In this example, I run the command and display the first ten lines that it produces:

$ mockgen -package mocks -source ../daos/people/DAO.go \
        -imports  'personModel=github.com/goblimey/films/models/person' | head -10
// Automatically generated by MockGen. DO NOT EDIT!
// Source: ../daos/people/DAO.go

package mocks

import (
    person "github.com/goblimey/films/models/person"
    dbsession "github.com/goblimey/films/utilities/dbsession"
    gomock "github.com/golang/mock/gomock"
)

**Comment by codyoss** [comment]
Hey @godolatry would you mind making a small repo that reproduces this?

**Comment by godolatry** [comment]
I reported this error more than three years ago.  It would have been a
whole lot easier to do what you’re asking back then.

Having given up with mock I found another Go mocking framework and used
that.  I found a couple of less important issues with that and the
maintainer fixed them within about a day.  If you want people to use this
package, you need to react a tad faster.

If I can find the time I will look through my bug report and try to figure
out what I did to provoke the error.  It looks pretty self explanatory.

Regards

Simon

**Comment by codyoss** [comment]
Sorry that it has taken so long. I have just started to takeover maintaining this package this week and I am trying to triage all open issues and PRs no matter how old.

If you have the time I would appreciate it, but if not I understand and will close this out in a couple of weeks as it does not seem to have too many upvotes and I wonder if it might have been fixed since you last used the package. Thanks.
